forum/profiles/tests.py

```diff
@@ -618,15 +622,179 @@ def test_save_startup_wrong_startup_id(self):
+            user=self.user1,
+            company_name=self.startup1_name,
+            industry=self.startup1_industry,
+            size=self.startup1_size,
+            country=self.startup1_country,
+            city=self.startup1_city,
+            zip_code='90002',
+            address='Some street 7',
+            phone='+380632225577',
+            email='random@email.com',
+            description='Some description',
+        )
+        self.user2 = User.objects.create_user(password='password2', email='user2@email.com')
+        self.startup2 = StartupProfile.objects.create(
+            user=self.user2,
+            company_name='Small Business',
+            industry='tourism',
+            size='200',
+            country='USA',
+            city='London',
+            zip_code='E1 6AN',
+            address='Some street 74',
+            phone='+380632225522',
+            email='randdsaom@email.com',
+            description='Some description',
+        )
+
+        # 3rd user will be acting as investor
+        self.user3 = User.objects.create_user(password='password3', email='user3@email.com')
+        self.investor = InvestorProfile.objects.create(
+            user=self.user3,
+            country="Ukraine",
+            phone="+380631234455",
+            email="testca3se@gmail.com",
+        )
+
+        # Adding startups to favourites
+        startups = [self.startup1, self.startup2]
+        self.investor.followed_startups.add(*startups)
+
+        # Generating user's access token
+        self.token_user3 = self.get_jwt_token(self.user3)
+
+    def get_jwt_token(self, user):
+        """Helper method to create a JWT token for a user."""
+        refresh = RefreshToken.for_user(user)
+        return str(refresh.access_token)
+
+    def test_get_followed_startups_anonymous(self):
+        """Test that checks if anonymous user has no access to the protected endpoint"""
+        url = reverse(self.get_saved_startups_url)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_get_followed_startups(self):
+        """Test that checks if authorized user can get list of startups"""
+        url = reverse(self.get_saved_startups_url)
+        self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {self.token_user3}')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+
+    def search_and_assert(self, search_query, expected_count: int):
+        """Helper method to perform search and assert results."""
+        url = reverse(self.get_saved_startups_url) + f'?search={search_query}'
+        self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {self.token_user3}')
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), expected_count)
```

Comment:
Enhance assertions by validating that the response contains the expected data, rather than just counting the results.

Reply:
done